tests/test_sparse_array.py

```diff
@@ -514,6 +514,105 @@ def test_data_conversion(cropped_multi_frames_v1, cropped_multi_frames_v2,
         assert array.num_frames_per_scan == 2
 
 
+def test_advanced_indexing(sparse_array_small, full_array_small):
+    array = sparse_array_small
+    full = full_array_small
+
+    # First test advanced integer indexing
+    to_test = [
+        [[0]],
+        [[1], [14]],
+        [[10], [-1]],
+        [[10, 7, 3], [7, 0, 9]],
+        [[3, 2, 1], [0, 1, 2]],
+        [[-1, 4, -3], [-3, 8, 7]],
+        [[7], 4],
+        [3, [8]],
+        [[-1, 4, -3], slice(2, 18, 2)],
+        [3, 4, [3, 2, 1], [9, 8, 7]],
+        [slice(None), slice(None), [5, 45, 32], [-3, -7, 23]],
+        [slice(None), slice(2, 18, 2), slice(0, 100, 3), [-5, 1, 2]],
+    ]
+
```

Comment:
@ercius You can see some of the advanced integer indexing test cases we are running above.

Reply:
these all look like possible user cases to me. I think this will make our code much more compatible with other applications like py4DSTEM.

Reply:
Sounds great!